autoscheduler/scheduler/views.py

```diff
@@ -1,4 +1,89 @@
-# Once we start adding routes, uncomment this line
-# from django.shortcuts import render
+import json
+from typing import List, Tuple
+from rest_framework.views import APIView
+from rest_framework.response import Response
 
-# Create your views here.
+from scheduler.create_schedules import create_schedules
+from scheduler.utils import UnavailableTime, CourseFilter
+from scraper.management.commands.scrape_courses import convert_meeting_time
+from scraper.serializers import SectionSerializer
+from scraper.models import Section
+
+def _parse_course_filter(course) -> CourseFilter:
+    """ Parses the given course to retrieve and convert it to a CourseFilter object
+        to be used in create_schedules
+    """
+
+    # Can assume subject & course_num will be given as strings
+    subject = course["subject"]
+    course_num = course["courseNum"]
+
+    sections = course["sections"]
+
+    honors = course["honors"]
+    web = course["web"]
+
+    return CourseFilter(subject=subject, course_num=course_num, section_nums=sections,
+                        honors=honors, web=web)
+
+def _parse_unavailable_time(avail) -> UnavailableTime:
+    """ Parses an availability input and convert it to an UnavailableTime object
+        to be used in create_schedules
+    """
+
+    start_time = convert_meeting_time(avail["startTime"])
+    end_time = convert_meeting_time(avail["endTime"])
+    day = avail["day"]
+
+    return UnavailableTime(start_time, end_time, day)
+
+
+def _serialize_schedules(schedules: List[Tuple[str]]) -> List[List]:
+    """ Converts the given schedules, retrieves the corresponding sections,
+        then serializes and returns them
+
+    Args:
+        schedules: The schedules returned from create_schedules
+
+    Returns
+        The list of given schedules with the corresponding serialized sections
+    """
+
+    # Retrieve the section models in bulk so we only do one DB query
+    # Put the section ids in a set to remove duplicates
+    section_set = set(section_id for schedule in schedules for section_id in schedule)
+    models = Section.objects.filter(id__in=section_set)
+
+    # Maps each section's id to their corresponding section model
+    sections_dict = {section.id: section for section in models.iterator()}
+
+    def sections_for_schedule(schedule):
+        sections = (sections_dict[section] for section in schedule)
+
+        return SectionSerializer(sections, many=True).data
+
+    return [sections_for_schedule(schedule) for schedule in schedules]
+
+class ScheduleView(APIView):
+    """ Handles requests to the generate schedules algorithm  """
+
+    def post(self, request): # pylint: disable=too-many-locals

```

Comment:
```suggestion
    def post(self, request):
```
There are only 7 locals so this linter rule isn't actually being broken